dev/docs/source/worksheet.rst

```diff
@@ -367,10 +367,12 @@ parameter is optional but when present is should be a valid
 :ref:`Format <format>` object.
 
 If required, it is also possible to specify the calculated result of the
-formula. This is occasionally necessary when working with non-Excel
-applications that don't calculate the result of the formula::
+formula (see discussion of formulas and the ``value`` parameter for the
+``write_formula()`` method above). However, using this parameter only writes
+a single value to the upper left cell in the result array and thus is largely
+unsuitable except for single-cell ranges::
```

Comment:
> and thus is largely unsuitable except for single-cell ranges

That isn't strictly true (the type of true that I think you are most interested in).

It still has a use in a multi-cell array function to set the formula result for the first cell. The formula result for the other cells can be set with `write_number()` (or whatever). So maybe the text could say something like:

```rst
However, using this parameter only writes a single value to the
upper left cell in the result array. For a multi-cell array
formula where the results are required, the other result values
can be specified by using ``write_number()`` to write to the appropriate
cell::

    # Specify the result for a single cell range.
    worksheet.write_array_formula('A1:A1', '{=SUM(B1:C1*B2:C2)}', format, 105)

    # Specify the results for a multi cell range.
    worksheet.write_array_formula('A1:A3', '{=TREND(C1:C3,B1:B3)}', format, 105)
    worksheet.write_number('A2', 102, format) 
    worksheet.write_number('A3', 104, format) 
```

If you feel up to it you can rebase and resubmit the patch. If not I can just merge it as is and make the change later. Let me know.

P.S. A better solution would be to support a list of lists as the result/value parameter. However, in 5 years of emails no-one has ever asked for or about that. In fact I suspect that even for ordinary formulas the result/value parameter is rarely used by end users.